File: ATMApplication/ATMAPI/Models/Transaction.cs

using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Threading.Tasks;

namespace ATMAPI.Models
{
    public class Transaction
    {
        [Key]
        public int Id { get; set; }

        [Required]
        public long BankAccountNoFrom { get; set; }
        [Required]
        public long BankAccountNoTo { get; set; }

        public enum TransactionType { Deposit, Withdrawal };

        public  TransactionType TransactionTypes { get; set; }

        [Required]
        public decimal TransactionAmount { get; set; }

        [Required]
        public DateTime TransactionDate { get; set; }
    }
}

File: ATMFrontEnd/ATMApplication/src/app/core/model/transaction.ts
export class Transaction {
    id: number;
    bankAccountNoFrom: number;
    bankAccountNoTo: number;
    transactionTypes: number;
    transactionAmount:number;
    transactionDate: Date = new Date();
    constructor(
     id: number,
    bankAccountNoFrom: number,
    bankAccountNoTo: number,
    transactionType: number,
    transactionAmount:number,
    transactionDate:Date) {
         this.id = id;
         this.bankAccountNoFrom = bankAccountNoFrom;
         this.bankAccountNoTo = bankAccountNoTo;
         this.transactionTypes =  transactionType;
         this.transactionAmount = transactionAmount;
         this.transactionDate = transactionDate;
    }
}